table/tables/tables.go

```diff
@@ -999,7 +999,7 @@ func (t *TableCommon) RemoveRecord(ctx sessionctx.Context, h kv.Handle, r []type
 		return err
 	}
 
-	if shouldWriteBinlog(ctx) {
+	if shouldWriteBinlog(ctx) && !t.meta.IsCommonHandle {
```

Comment:
How about we change `shouldWriteBinlog` to add the table meta as an argument?
Then we will be always consistent.